core/vdbe/mod.rs

```diff
@@ -1434,6 +1434,11 @@ impl Program {
                                 );
                                 state.registers[*dest] = result;
                             }
+                            ScalarFunc::ConcatWs => {
+                                let start_reg = *start_reg;
+                                let result = exec_concat_ws(&state.registers[start_reg..]);
+                                state.registers[*dest] = result;
+                            }
```

Comment:
I think you must use FuncCtx to retrieve the number of registers so that 
```
state.registers[start_reg..]
```
has some upper bound. If not it will try to get ALL registers that are allocated after `start_reg` causing wrong results

Reply:
OH, thanks. I'll apply it as well.